controllers/grid/settings/user/UserGridHandler.php

```diff
@@ -203,7 +214,17 @@ protected function loadData($request, $filter)
         $collector->limit($rangeInfo->getCount());
         $collector->offset($rangeInfo->getOffset() + max(0, $rangeInfo->getPage() - 1) * $rangeInfo->getCount());
         $iterator = $collector->getMany();
-        return new VirtualArrayIterator(iterator_to_array($iterator, true), $totalCount, $rangeInfo->getPage(), $rangeInfo->getCount());
+        $users = iterator_to_array($iterator, true);
+        foreach ($users as $user) {
+            $roles = array();
+            $userGroupsIterator = Repo::userGroup()->userUserGroups($user->getId(), $context->getId());
+            $userGroups = iterator_to_array($userGroupsIterator);
+            foreach ($userGroups as $userGroup) {
+                $roles[] = $userGroup->getLocalizedName();
+            }
+            $user->setData('roles', join(', ', $roles));
```

Comment:
Still about this line, I think `setData('roles')` might be dangerous, I see the potencial to raise conflicts in the future.

So, I think the right solution is to use another type of "cell provider" on the `$this->addColumn()`, probably the `ColumnBasedGridCellProvider`. See an example here:
https://github.com/pkp/pkp-lib/blob/777b10995699b8419a61f62adbceb5cfcba8522d/classes/controllers/grid/feature/selectableItems/ItemSelectionGridColumn.php#L38

And the implementation here:
https://github.com/pkp/pkp-lib/blob/777b10995699b8419a61f62adbceb5cfcba8522d/classes/controllers/grid/feature/selectableItems/ItemSelectionGridColumn.php#L73-L80

To avoid creating a new file/class just for this, you can make use of a dynamic class, which would look something like this:
```php
$columnBasedGridCellProvider = new ColumnBasedGridCellProvider();
$this->addColumn(
    new class (
        'roles',
        'user.roles',
        null,
        null,
        $columnBasedGridCellProvider
    ) extends GridColumn {
        public function getTemplateVarsFromRow($row) {
            //implementation
        }
    }
);
```

Reply:
Hi @jonasraoni, thanks for all the detailled review comments, it's much appreciated!

I did every requested change and I hope it aligns more with the standard of OJS this way. =)

Reply:
Thanks for your patience 🤗